docs/source/api/user_current.rst

```diff
@@ -147,49 +150,99 @@ Request Structure
 	User-Agent: curl/7.47.0
 	Accept: */*
 	Cookie: mojolicious=...
-	Content-Length: 483
+	Content-Length: 465
 	Content-Type: application/json
 
 	{ "user": {
-		"addressLine1": "not a real address",
-		"addressLine2": "not a real address either",
-		"city": "not a real city",
-		"company": "not a real company",
-		"country": "not a real country",
-		"email": "not@real.email",
-		"fullName": "Not a real fullName",
-		"phoneNumber": "not a real phone number",
-		"postalCode": "not a real postal code",
-		"publicSshKey": "not a real ssh key",
-		"stateOrProvince": "not a real state or province",
-		"tenantId": 1,
+		"addressLine1": null,
```

Comment:
do we really want users to give null in the request? I really dislike that of a user of any API. As a user do I need to define the key and pass in null? Is that the same behavior as omitting the key? How do various programming languages clients handle setting `null` fields.

But maybe it is a common approach in TO which in that case fine leave it but I strongly side with removing null in requests/responses going forward

Reply:
In order for `PUT` to remain idempotent, all fields ought to be defined in every request. We could treat missing fields as implicitly `null`, but that introduces problems when you add fields to a request. Because old clients won't know about new fields they'll `PUT` without them and accidentally set something to `null`. The Perl acted as a `PATCH` by only updating the fields it found in the request, but as that was breaking with the consistency of our API  - and seemingly RFC non-compliant - I elected to change the behavior.

Reply:
`but that introduces problems when you add fields to a request. Because old clients won't know about new fields they'll PUT without them and accidentally set something to null` API versioning alleviate this concern?

Reply:
Yes and no. In general, yes, but because the old Perl version of the endpoint didn't require fields to be defined and I didn't feel like rewriting it that way was a good idea, it seems safer to just return an error rather than use implicit `null`s, because clients coded against the Perl version of APIs 1.1 to 1.3 won't expect anything to have changed. Short of implementing the behavior unchanged failing fast I think failing fast is the best option.

Now that I'm saying all this, I think maybe I ought to bring that up to the mailing list, seems like a big change for me to make with minimal community input.